### PR TITLE
Fix: prevent of forwarding incorrect types to Lua

### PR DESCRIPTION
**Pull request by ChugunovRoman**

### CODE REVIEW

File: src/xrServerEntities/xrServer_Objects_ALife_Monsters.cpp [hunk 1]
```diff
@@ -473,6 +473,8 @@ void CSE_ALifeTraderAbstract::set_specific_character(shared_str new_spec_char)
 
 void CSE_ALifeTraderAbstract::set_character_profile(shared_str new_profile) { m_sCharacterProfile = new_profile; }
 shared_str CSE_ALifeTraderAbstract::character_profile() { return m_sCharacterProfile; }
+void CSE_ALifeTraderAbstract::set_character_name(shared_str new_name) { m_character_name = new_name.c_str(); }
+shared_str CSE_ALifeTraderAbstract::character_name() { return m_character_name.c_str(); }
```
Comment (Xottab-DUTY) [inline src/xrServerEntities/xrServer_Objects_ALife_Monsters.cpp]: Please, don't introduce new functions just to use them in script export. there are lambdas for that.

Also, you call `c_str()` here, but function still returns `shared_str`.

File: src/xrServerEntities/xrServer_Objects_ALife_Monsters.cpp [hunk 2]
```diff
@@ -473,6 +473,8 @@ void CSE_ALifeTraderAbstract::set_specific_character(shared_str new_spec_char)
 
 void CSE_ALifeTraderAbstract::set_character_profile(shared_str new_profile) { m_sCharacterProfile = new_profile; }
 shared_str CSE_ALifeTraderAbstract::character_profile() { return m_sCharacterProfile; }
+void CSE_ALifeTraderAbstract::set_character_name(shared_str new_name) { m_character_name = new_name.c_str(); }
```
Comment (Xottab-DUTY) [inline src/xrServerEntities/xrServer_Objects_ALife_Monsters.cpp]: This is just messing around with pcstr -> shared_str -> pcstr -> shared_str conversions. What's the point this function?

**Review by Xottab-DUTY — CHANGES_REQUESTED** [review]
What's the crash log?

**Comment by ChugunovRoman** [comment]
> What's the crash log?

@Xottab-DUTY 
```
! SCRIPT ERROR: std::runtime_error: 'Trying to use unregistered class: : NSt7__cxx1112basic_stringIcSt11char_traitsIcE6xallocIcEEE'
 
FATAL ERROR
 
[error] Expression    : LUA error
[error] Function      : onErrorCallback
[error] File          : /home/ruut/Documents/qemu_shared/xray-16/src/xrScriptEngine/script_engine.cpp
[error] Line          : 804
[error] Description   : std::runtime_error: 'Trying to use unregistered class: : NSt7__cxx1112basic_stringIcSt11char_traitsIcE6xallocIcEEE'
 

stack trace:

xrDebug::GatherInfo(char*, unsigned long, ErrorLocation const&, char const*, char const*, char const*, char const*)
xrDebug::Fail(bool&, ErrorLocation const&, char const*, char const*, char const*, char const*)
CScriptEngine::onErrorCallback(lua_State*, char const*, int, char const*)
CScriptEngine::lua_pcall_failed(lua_State*)
/home/ruut/Documents/qemu_shared/xray-16/bin/x64/Debug/xrLuajit-debug.so(+0x65ee5) [0x7f03735a9ee5]
/home/ruut/Documents/qemu_shared/xray-16/bin/x64/Debug/xrLuajit-debug.so(+0x3af4f) [0x7f037357ef4f]
/home/ruut/Documents/qemu_shared/xray-16/bin/x64/Debug/xrLuajit-debug.so(+0x3b6b9) [0x7f037357f6b9]
luabind::detail::function_object_impl<std::__cxx11::basic_string<char, std::char_traits<char>, xalloc<char> > (*)(CSE_ALifeTraderAbstract*, char const*), luabind::meta::type_list<std::__cxx11::basic_string<char, std::char_traits<char>, xalloc<char> >, CSE_ALifeTraderAbstract*, char const*>, luabind::meta::type_list<> >::entry_point(lua_State*)
/home/ruut/Documents/qemu_shared/xray-16/bin/x64/Debug/xrLuajit-debug.so(+0x65ee5) [0x7f03735a9ee5]
/home/ruut/Documents/qemu_shared/xray-16/bin/x64/Debug/xrLuajit-debug.so(lua_pcall+0xae) [0x7f037355da7e]
luabind::detail::pcall(lua_State*, int, int)
void luabind::detail::call_member_impl<void, luabind::meta::type_list<>>(lua_State*, std::integral_constant<bool, true>, luabind::meta::index_list<>)
void luabind::wrap_base::call<void>(char const*) const
void luabind::call_member<void>(luabind::wrap_base const*, char const*)
CWrapperAbstractCreature<CSE_ALifeCreatureActor>::on_register()
CALifeUpdateManager::new_game(char const*)
CALifeUpdateManager::load(char const*, bool, bool)
CALifeSimulator::CALifeSimulator(IPureServer*, shared_str*)
CALifeSimulator* xr_new<CALifeSimulator, xrServer*, shared_str*>(xrServer*&&, shared_str*&&)
game_sv_Single::Create(shared_str&)
```

I'll correcting the comments and add changes of other Lua methods to this PR.

**Comment by ChugunovRoman** [comment]
Sorry for the trash in the code. I was in a hurry. It was only necessary to remove the return for the set_character_name method.

**Review by Xottab-DUTY — APPROVED** [review]

**Pull request merged**